### DISCUSSION TITLE
Memory usage optimisation

### DISCUSSION THREAD

**Discussion by frutik** [qdrant/qdrant#586] [Q&A] [answered]
Hi, 

I am trying to evaluate the Qdrant as a replacement for an annoy-based custom solution. My dataset consists of 2.5 documents containing vectors of dimension 512, floats + some JSON payload.  Actually, RAM consumption of quadrant is huge, compared to Annoy. Annoy based app is running on 1GB ram, Qdrant is dying (probably hits some limits of my test machine) during the attempt to load the dataset on my laptop. Any suggestions on some (undocumented) resources usage tweaks?

**Comment by generall** [comment, marked as answer]
Hi @frutik, it is possible to lower the memory usage if you don't need production-grade speed at the moment. It this case what you can do is disable HNSW indexing throw the configuration:  https://github.com/qdrant/qdrant/blob/master/config/config.yaml#L48 (put large enough data, and index will never be created).

Unfortunately, memory consumption is one of the downsides of HNSW - it requires random access to stored vectors in order to make the search fast and accurate. At the same time - HNSW is the best candidate to combine it with payload filtering functionality, not many other index types are suitable for this, and Annoy is not one of them as well. 

Additionally, the payload currently is also stored in memory, but we are going to change that in following release: https://github.com/qdrant/qdrant/issues/406

**Reply by frutik** [reply]
> and Annoy is not one of them as well.

the main reason for me to search for the alternative.

> it is possible to lower the memory usage if you don't need production-grade speed at the moment

well, that makes sense, thank you. my use-case is offline processing, so reasonably slow search is not a problem
